### DISCUSSION TITLE
Background profile not loading when a saved roast profile is opened / loaded.

### DISCUSSION THREAD

**Discussion by Kivbean** [artisan-roaster-scope/artisan#911] [General]
When we open up a saved  roast profile, the background profile is not loading with it. If you open the profile background selection box, no roast file is showing up in the loaded/select input box at the bottom.
Any help would be gratefully received.
thanks
Kiv

**Comment by poundy** [comment]
If you can attach those profiles here, we can take a look. But if you want to dig in, the roast is stored inside the ALOG file, which is just a plain text file with structure.... so open up a copy of one in the text editor of your choice and do a search for "background" (there are two instances of this in an example I looked at - "backgroundpath" and "backgroundUUID") and you should see immediately following the labels what the saved values are. If it has nothing in there, then when the profile was saved, it was not saved with a background loaded. If it has a background path listed, then it's possible the background file is no longer found at that path - I'd check the paths are as you'd expect

**Reply by jessemedley** [reply]
I am now also having this problem after updating Artisan on Linux. The ALOG files list the "backgroundpath" and "backgroundUUID" files and the files are located in the right place.
I assume that this is a permission problem, but I haven't been able to solve it.

**Comment by MAKOMO** [comment]
A background profile is loaded if its name is displayed as subtitle above the upper right corner of the graph in Artisans main window. This subtitle is rendered in grey if the background profile is hidden. Double click on that subtitle to un-hide it. Double click on it again to hide it again.

The option `Always hide background when loading a profile` of the background dialog (menu Roast >> Background) instructs Artisan to hide a background profile associated with the profile that is loaded after loading it automatically. Untick this option if you want the background profile to be displayed and not hidden on profile load.

Note that a background profile referenced by a profile is not dereferenced directly by the given profile path, but by its UUID (a unique identifier). Artisan maintains a database associating those profile UUIDs to file paths to be able to dereference them for loading. The entry of a profile within that data base is updated each time a profile is successfully loaded such that Artisan always "knows" its last valid location.

If on loading a profile the referenced background profiles UUID cannot be dereferenced using this database (maybe because it was never loaded on that account by Artisan) or if Artisan lacks the rights to access this dereferenced file, loading of this background profile fails.

**Reply by jessemedley** [reply]
thank you,
the "Always hide background when loading a profile" option was ticked.
